scipy/sparse/linalg/isolve/iterative.py

```diff
@@ -524,6 +526,16 @@ def gmres(A, b, x0=None, tol=1e-5, restart=None, maxiter=None, M=None, callback=
     True
     """
 
+    # Faster restarted GMRES based on modified Gram-Schmidt orthogonalization
+    if orth == 'mgs':
+        if atol is None:
+            x, info = _gmres_mgs(A, b, x0=x0, restart=restart, tol=tol, maxiter=maxiter, M=M, 
+                                 callback=callback, restrt=restrt)
+            return x, info
+        else:
+            warnings.warn("The option 'atol' was deprecated with 'orth == \'mgs\''",
```

Comment:
I am not sure about saying `atol` is deprecated. Also there is this issue which got reported #13935.

Reply:
Dear @tupui 
I added the additional option parameter `orth='mgs'` to start the new GMRES algorithm, which hasn't the option `atol` when explicitly using the keyword `orth='mgs'`. But if one want to use `atol`, just don't specify this parameter `orth=mgs` (that is `orth=None`), at this time, we will run the original GMRES algorithm (`atol` exists in the original GMRES)

Reply:
I don't understand what modified Gram-Schmidt orthogonalization has to do with the stopping criterion. @zhaong6 Can you give some details?

Reply:
No, for the new GMRES implementation. It has nothing to do with convergence criterion. If specifying the keyword `orth=mgs`, this is only a new implementation (`atol` is deprecated in the  new implementation because most softwares don't care `atol` (absolute residual))

Reply:
Using the keyword `orth` is to make a distinction between the new algorithm ('orth='mgs'') and the original algorithm ('orth=None') for compatibility, it has nothing to do with criterions you used.

Reply:
> I added the additional option parameter `orth='mgs'` to start the new GMRES algorithm, which hasn't the option `atol` when explicitly using the keyword `orth='mgs'`. But if one want to use `atol`, just don't specify this parameter `orth=mgs` (that is `orth=None` by default), at this time, we will run the original GMRES algorithm (`atol` exists in the original GMRES)

What I am pointing out is that `atol` is not deprecated, the terms is incorrect. It should say that it's not used in this case. If we want to deprecate the other algorithm (and then `atol` would be deprecated) that's another story and we need to be careful here.

As for implementing `atol` or not, I leave this to specialists (which I am not here).

Reply:
Okay, my bad. This terms "The option `atol`  was deprecated with `orth='mgs'`" in line 536 is indeed incorrect because it is only not used in the new implementation with 'orth='mgs''

Reply:
By "new implementation" you mean this PR?

Reply:
Dear @tupui ,
I have modified this incorrect expression and summitted this PR again. Thank you for your suggestions and corrections.

Reply:
> By "new implementation" you mean this PR?

@nschloe Yes.